File: Tarea8/libreria.java
package Tarea8;

import java.util.ArrayList;
import java.util.List;

class libreria {
    List<usuarios> usuarios = new ArrayList<>();
    List<libros> libros = new ArrayList<>();

    public void registrarUsuario(usuarios usuario) {
        usuarios.add(usuario);
    }

    public void registrarLibro(libros libro) {
        libros.add(libro);
    }

    public void rentarLibro(usuarios usuario, libros libro) {
        if (!libro.EsRentado()) {
            libro.setEsRentado(true);
            usuario.agregarLibroRentado(libro);
        } else {
            System.out.println("El libro ya está rentado.");
        }
    }

    public void listarUsuarios() {
        for (usuarios usuario : usuarios) {
            System.out.println(usuario);
        }
    }

    public void listarLibros() {
        for (libros libro : libros) {
            System.out.println(libro);
        }
    }

    public void listarUsuariosConLibros() {
        for (usuarios usuario : usuarios) {
            if (!usuario.getLibrosRentados().isEmpty()) {
                System.out.println(usuario);
            }
        }
    }

    public void listarLibrosNoRentados() {
        for (libros libro : libros) {
            if (!libro.EsRentado()) {
                System.out.println(libro);
            }
        }
    }

    public void listarLibrosRentados() {
        for (libros libro : libros) {
            if (libro.EsRentado()) {
                System.out.println(libro);
            }
        }
    }
}

File: Tarea8python/libreria.py
class libreria:
    def __init__(self):
        self.usuarios = []
        self.libros = []

    def registrar_usuario(self, usuario):
        self.usuarios.append(usuario)

    def registrar_libro(self, libro):
        self.libros.append(libro)

    def rentar_libro(self, usuario, libro):
        if not libro.es_rentado:
            libro.es_rentado = True
            usuario.agregar_libro_rentado(libro)
        else:
            print("El libro ya está rentado.")

    def listar_usuarios(self):
        for usuario in self.usuarios:
            print(usuario)

    def listar_libros(self):
        for libro in self.libros:
            print(libro)

    def listar_usuarios_con_libros(self):
        for usuario in self.usuarios:
            if usuario.libros_rentados:
                print(usuario)

    def listar_libros_no_rentados(self):
        for libro in self.libros:
            if not libro.es_rentado:
                print(libro)

    def listar_libros_rentados(self):
        for libro in self.libros:
            if libro.es_rentado:
                print(libro)